bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -0,0 +1,1153 @@
+                    if (Objects.nonNull(putResourceId)) {
+                        putResource = new Resource(ResourceType.SCENE).setRecallAction(RecallAction.ACTIVE);
+                    }
+                }
+                break;
+
+            case CHANNEL_2_DYNAMICS:
+                Duration clearAfter = Duration.ZERO;
+                if (command instanceof QuantityType<?>) {
+                    QuantityType<?> durationMs = ((QuantityType<?>) command).toUnit(MetricPrefix.MILLI(Units.SECOND));
+                    if (Objects.nonNull(durationMs) && durationMs.longValue() > 0) {
+                        Duration duration = Duration.ofMillis(durationMs.longValue());
+                        dynamicsDuration = duration;
+                        dynamicsExpireTime = Instant.now().plus(DYNAMICS_ACTIVE_WINDOW);
+                        clearAfter = DYNAMICS_ACTIVE_WINDOW;
+                        logger.debug("{} -> handleCommand() dynamics setting {} valid for {}", resourceId, duration,
+                                clearAfter);
+                    }
+                }
+                cancelTask(dynamicsResetTask, false);
+                dynamicsResetTask = scheduler.schedule(() -> clearDynamicsChannel(), clearAfter.toMillis(),
+                        TimeUnit.MILLISECONDS);
+                return;
+
+            default:
+                logger.warn("{} -> handleCommand() channelUID:{} not supported", resourceId, channelUID);
+                return;
+        }
+
+        if (putResource == null) {
+            logger.warn("{} -> handleCommand() command:{} not supported on channelUID:{}", resourceId, command,
+                    channelUID);
+            return;
+        }
+
+        putResourceId = Objects.nonNull(putResourceId) ? putResourceId : commandResourceIds.get(putResource.getType());
+        if (putResourceId == null) {
+            logger.warn("{} -> handleCommand() channelUID:{}, command:{}, putResourceType:{} => missing resource ID",
+                    resourceId, channelUID, command, putResource.getType());
+            return;
+        }
+
+        if (DYNAMIC_CHANNELS.contains(channelId)) {
+            if (Instant.now().isBefore(dynamicsExpireTime) && !dynamicsDuration.isZero()
+                    && !dynamicsDuration.isNegative()) {
+                if (ResourceType.SCENE == putResource.getType()) {
+                    putResource.setRecallDuration(dynamicsDuration);
+                } else {
+                    putResource.setDynamicsDuration(dynamicsDuration);
+                }
+            }
+        }
+
+        putResource.setId(putResourceId);
+        logger.debug("{} -> handleCommand() put resource {}", resourceId, putResource);
+
+        try {
+            getBridgeHandler().putResource(putResource);
+        } catch (ApiException | AssetNotLoadedException e) {
+            logger.warn("{} -> handleCommand() error {}", resourceId, e.getMessage(), e);
```

Comment:
Can this warning be made more user-oriented? Also, it would be better to not log full stack trace, but instead make that conditional on debug level, i.e.:
```suggestion
            if (logger.isDebugEnabled()) {
                logger.debug("{} -> handleCommand() error {}", resourceId, e.getMessage(), e);
            } else {
                logger.warn("Failed to send command {} for resource {}: {}", command, resourceId, e.getMessage());
            }
```

Reply:
Ok. Done.